lib/action_cable/connection/base.rb

```diff
@@ -180,6 +180,11 @@ def allow_request_origin?
           end
         end
 
+        def allowed_origins_match? origin
+          allowed_origins = Array(server.config.allowed_request_origins)
+          allowed_origins.any? { |allowed_origin| allowed_origin.is_a?(Regexp) ? allowed_origin =~ origin : allowed_origin == origin }
```

Comment:
Actually, given this, we don't need the extra method. We can just inline this === check like we had the original include call.